app/com/linkedin/drelephant/mapreduce/data/MapReduceTaskData.java

```diff
@@ -28,13 +28,17 @@
   private long _totalTimeMs = 0;
   private long _shuffleTimeMs = 0;
   private long _sortTimeMs = 0;
+  private long _startTimeMs = 0;
+  private long _finishTimeMs = 0;
   private boolean _sampled = false;
 
   public MapReduceTaskData(MapReduceCounterData counterHolder, long[] time) {
```

Comment:
I realized there will a lot of refactoring involved in changing this. ( All Heuristics, tests etc.). I'll send a different PR for refactoring.